index.js

```diff
@@ -179,7 +179,7 @@ Request.prototype.run = function () {
                 }
             }
 
-            if (res.status >= 400) {
+            if (!res.status || res.status < 100 || res.status >= 400) {
```

Comment:
@gwicke IMHO a general 500 with message like `Response status parse error` could be fine, or we could mimic node's behaviour and throw 504, however the choice is not clear to me, as this has nothing to do with a `Gateway Timeout`